pkg/config/config.go

```diff
@@ -97,20 +103,34 @@ func fetchConfig(ctx context.Context, r repositories, owner, repo, path string,
 }
 
 type repositories interface {
+	Get(context.Context, string, string) (*github.Repository,
+		*github.Response, error)
 	GetContents(context.Context, string, string, string,
 		*github.RepositoryContentGetOptions) (*github.RepositoryContent,
 		[]*github.RepositoryContent, *github.Response, error)
 }
 
 // IsEnabled determines if a repo is enabled by interpreting the provided
 // org-level and repo-level OptConfigs.
-func IsEnabled(o OrgOptConfig, r RepoOptConfig, repo string) bool {
+func IsEnabled(ctx context.Context, o OrgOptConfig, r RepoOptConfig, rep repositories, owner, repo string) (bool, error) {
```

Comment:
It's not good to have a public function with a parameter of a private type. Also, all the functions/packages only pass around `github.Client` in code currently. Make `IsEnabled()` take a `github.Client` then simply call a private `isEnabled()` which takes a `repositories`, then change each policy to pass in the client, not client.Repositories. The tests should then call the private `isEnabled` with the `repositories` mock object.

Reply:
It makes sense. Updated with a57b1ebf04d4dd118fb3cf5b7fa83a44a544f455.